packages/open-api-gateway/src/project/open-api-gateway-java-project.ts

```diff
@@ -0,0 +1,253 @@
+import { ClientSettings } from "./codegen/components/client-settings";
+import { DocsProject } from "./codegen/docs-project";
+import { generateClientProjects } from "./codegen/generate";
+import { GeneratedJavaClientProject } from "./codegen/generated-java-client-project";
+import { ClientLanguage } from "./languages";
+import { getJavaSampleSource, JavaSampleCodeOptions } from "./samples/java";
+import { OpenApiSpecProject } from "./spec/open-api-spec-project";
+import { OpenApiGatewayProjectOptions } from "./types";
+
+const OPENAPI_GATEWAY_PDK_PACKAGE_NAME =
+  "software.aws.awsprototypingsdk/open-api-gateway@^0";
+
+/**
+ * Configuration for the OpenApiGatewayJavaProject
+ */
+export interface OpenApiGatewayJavaProjectOptions
+  extends JavaProjectOptions,
+    OpenApiGatewayProjectOptions {}
+
+/**
+ * Synthesizes a Java Project with an OpenAPI spec, generated clients, a CDK construct for deploying the API
+ * with API Gateway, and generated lambda handler wrappers for type-safe handling of requests.
+ *
+ * @pjid open-api-gateway-java
+ */
+export class OpenApiGatewayJavaProject extends JavaProject {
+  /**
+   * A reference to the generated java client
+   */
+  public readonly generatedJavaClient: JavaProject;
+
+  /**
+   * References to the client projects that were generated, keyed by language
+   */
+  public readonly generatedClients: { [language: string]: Project };
+
+  /**
+   * The directory in which the OpenAPI spec file(s) reside, relative to the project srcdir
+   */
+  public readonly specDir: string;
+
+  /**
+   * The directory in which the api generated code will reside, relative to the project srcdir. This will also be used
+   * as the package for the api project.
+   */
+  public readonly apiSrcDir: string;
+
+  /**
+   * The name of the spec file
+   */
+  public readonly specFileName: string;
+
+  /**
+   * The directory in which generated client code will be generated, relative to the outdir of this project
+   */
+  public readonly generatedCodeDir: string;
+
+  /**
+   * Force to generate code and docs even if there were no changes in spec
+   */
+  public readonly forceGenerateCodeAndDocs?: boolean;
+
+  private readonly hasParent: boolean;
+
+  constructor(options: OpenApiGatewayJavaProjectOptions) {
+    super({
+      ...options,
+      sample: false,
+    });
+
+    if (options.specFile) {
+      this.specDir = path.dirname(options.specFile);
+      this.specFileName = path.basename(options.specFile);
+    } else {
+      this.specDir = "spec";
+      this.specFileName = "spec.yaml";
+    }
+    this.generatedCodeDir = options.generatedCodeDir ?? "generated";
+    this.forceGenerateCodeAndDocs = options.forceGenerateCodeAndDocs ?? false;
+    this.apiSrcDir = options.apiSrcDir ?? "api";
+
+    // Generated project should have a dependency on this project, in order to run the generation scripts
+    [
+      OPENAPI_GATEWAY_PDK_PACKAGE_NAME,
+      "software.constructs/constructs@^10",
```

Comment:
java deps in projen are painful...

Reply:
Yeah it's a shame we have to specify a version :(